engine/apps/alerts/models/alert_group.py

```diff
@@ -1076,32 +1179,55 @@ def _bulk_acknowledge(user: User, alert_groups_to_acknowledge: "QuerySet[AlertGr
 
         # it is needed to unserolve those alert_groups which were resolved to build proper log.
-        alert_groups_to_unresolve_before_acknowledge = alert_groups_to_acknowledge.filter(resolved=True)
+        alert_groups_to_unresolve_before_acknowledge = alert_groups_to_acknowledge.filter(resolved=models.Value("1"))
 
         # it is needed to unsilence those alert_groups which were silenced to build proper log.
-        alert_groups_to_unsilence_before_acknowledge = alert_groups_to_acknowledge.filter(silenced=True)
+        alert_groups_to_unsilence_before_acknowledge = alert_groups_to_acknowledge.filter(silenced=models.Value("1"))
 
         # convert current qs to list to prevent changes by update
         alert_groups_to_acknowledge_list = list(alert_groups_to_acknowledge)
         alert_groups_to_unresolve_before_acknowledge_list = list(alert_groups_to_unresolve_before_acknowledge)
         alert_groups_to_unsilence_before_acknowledge_list = list(alert_groups_to_unsilence_before_acknowledge)
 
-        alert_groups_to_acknowledge.update(
-            acknowledged=True,
-            resolved=False,
-            resolved_at=None,
-            resolved_by=AlertGroup.NOT_YET,
-            resolved_by_user=None,
-            silenced_until=None,
-            silenced_by_user=None,
-            silenced_at=None,
-            silenced=False,
-            acknowledged_at=timezone.now(),
-            acknowledged_by_user=user,
-            acknowledged_by=AlertGroup.USER,
-            is_escalation_finished=True,
-        )
+        now = timezone.now()
+
+        previous_states = []
+        for alert_group in alert_groups_to_acknowledge_list:
+            # keep previous state information
+            previous_states.append(alert_group.state)
+
+            alert_group.acknowledged_at = now
+            alert_group.acknowledged = True
+            alert_group.resolved = False
+            alert_group.resolved_at = None
+            alert_group.resolved_by = AlertGroup.NOT_YET
+            alert_group.resolved_by_user = None
+            alert_group.silenced_until = None
+            alert_group.silenced_by_user = None
+            alert_group.silenced_at = None
+            alert_group.silenced = False
+            alert_group.acknowledged_by_user = user
+            alert_group.acknowledged_by = AlertGroup.USER
+            alert_group.is_escalation_finished = True
+
+        fields_to_update = [
+            "acknowledged",
+            "resolved",
+            "resolved_at",
+            "resolved_by",
+            "resolved_by_user",
+            "silenced_until",
+            "silenced_by_user",
+            "silenced_at",
+            "silenced",
+            "acknowledged_at",
+            "acknowledged_by_user",
+            "acknowledged_by",
+            "is_escalation_finished",
+        ]
+        AlertGroup.all_objects.bulk_update(alert_groups_to_acknowledge_list, fields_to_update, batch_size=1000)
```

Comment:
Here it's better to use `alert_groups_to_acknowledge.update` instead of `AlertGroup.all_objects.bulk_update` because regular queryset update is more efficient.

Regular update SQL will look similar to this:

```sql
UPDATE table
SET table.field = 1
WHERE table.id IN (1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
```

Batch update SQL will look similar to this:

```sql
UPDATE table
SET table.field = CASE
                      WHEN (table.id = 1) THEN 1
                      WHEN (table.id = 2) THEN 1
                      WHEN (table.id = 3) THEN 1
                      WHEN (table.id = 4) THEN 1
                      WHEN (table.id = 5) THEN 1
                      WHEN (table.id = 6) THEN 1
                      WHEN (table.id = 7) THEN 1
                      WHEN (table.id = 8) THEN 1
                      WHEN (table.id = 9) THEN 1
                      WHEN (table.id = 10) THEN 1
                      ELSE NULL END
WHERE table.id IN (1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
```